### PR TITLE
NTR "has life cycle stage" relation

### PR DESCRIPTION
**Pull request by ddooley**
And "collected during life cycle stage" relation. See https://github.com/oborel/obo-relations/issues/598

### CODE REVIEW

File: src/ontology/ro-edit.owl [hunk 1]
```diff
@@ -6523,6 +6525,28 @@ SubObjectPropertyOf(obo:RO_0017004 obo:RO_0002610)
...
 IrreflexiveObjectProperty(obo:RO_0017004)
 
+# Object Property: obo:RO_0017500 (has life cycle stage)
+
+AnnotationAssertion(obo:IAO_0000115 obo:RO_0017500 "A relationship between an organism and a stage it undergoes within its full life cycle."@en)
+AnnotationAssertion(dc:contributor obo:RO_0017500 <https://orcid.org/0000-0002-1373-1705>)
+AnnotationAssertion(dc:contributor obo:RO_0017500 <https://orcid.org/0000-0002-8844-9165>)
+AnnotationAssertion(dc:date obo:RO_0017500 "2023-08-09T18:37:32Z"^^xsd:dateTime)
+AnnotationAssertion(rdfs:comment obo:RO_0017500 "This relation accommodates social, behavioural and biological (developmental) stages."@en)
+AnnotationAssertion(rdfs:label obo:RO_0017500 "has life cycle stage"@en)
```
Comment (cthoyt) [inline src/ontology/ro-edit.owl:6535]: please add example usages for both

**Review by anitacaron — CHANGES_REQUESTED** [review]
Please add an example of usage for both. (making Charles' review official)

File: src/ontology/ro-edit.owl (hunk 1 above)
Comment (ddooley) [reply src/ontology/ro-edit.owl:6527]: Resolved for this relation; "collected during" relation how handled in separate pull request.

Comment (cthoyt) [reply src/ontology/ro-edit.owl:6527]: Great, thanks for updating this.

File: src/ontology/ro-edit.owl [hunk 2]
```diff
@@ -7063,19 +7090,19 @@ DLSafeRule(Annotation(rdfs:seeAlso <https://github.com/oborel/obo-relations/issu
...
 DLSafeRule(Annotation(rdfs:label "inferring direct positive reg edge from input to regulatory subfunction") Body(ObjectPropertyAtom(obo:RO_0002352 Variable(<urn:swrl#B>) Variable(<urn:swrl#C>)) ObjectPropertyAtom(obo:RO_0002333 Variable(<urn:swrl#A>) Variable(<urn:swrl#B>)) ObjectPropertyAtom(obo:RO_0002015 Variable(<urn:swrl#D>) Variable(<urn:swrl#C>)))Head(ObjectPropertyAtom(obo:RO_0002629 Variable(<urn:swrl#A>) Variable(<urn:swrl#D>))))
-DLSafeRule(Annotation(rdfs:label "effector input is compound function input") Body(ObjectPropertyAtom(obo:RO_0002233 Variable(<urn:swrl#eff>) Variable(<urn:swrl#in>)) ObjectPropertyAtom(obo:RO_0002025 Variable(<urn:swrl#mf>) Variable(<urn:swrl#eff>)))Head(ObjectPropertyAtom(obo:RO_0002233 Variable(<urn:swrl#mf>) Variable(<urn:swrl#in>))))
```
Comment (cthoyt) [inline src/ontology/ro-edit.owl:7066]: Can you reduce this spurious diff?

Comment (ddooley) [reply src/ontology/ro-edit.owl]: I think this is resolved now!

**Review by cthoyt — APPROVED** [review]

File: src/ontology/ro-edit.owl (hunk 2 above)
Comment (cthoyt) [reply src/ontology/ro-edit.owl]: LGTM :)

**Review by anitacaron — APPROVED** [review]
Technically approved. Still needs ontology review.

**Review by anitacaron — DISMISSED** [review]
Technically approved. Still needs ontology review.

File: src/ontology/ro-edit.owl [hunk 3]
```diff
@@ -6612,6 +6614,31 @@ AnnotationAssertion(oboInOwl:creation_date obo:RO_0017009 "2023-03-29T17:06:06Z"
...
+ObjectPropertyRange(obo:RO_0017500 obo:UBERON_0000105)
+
+# Object Property: obo:RO_0017501 (life cycle stage of)
+
+AnnotationAssertion(obo:IAO_0000115 obo:RO_0017501 "The inverse of \"has life cycle stage\"."@en)
+AnnotationAssertion(terms:contributor obo:RO_0017501 <https://orcid.org/0000-0002-8844-9165>)
+AnnotationAssertion(terms:date obo:RO_0017501 "2023-08-09T18:40:21Z"^^xsd:dateTime)
+AnnotationAssertion(rdfs:label obo:RO_0017501 "life cycle stage of"@en)
+SubObjectPropertyOf(obo:RO_0017501 obo:RO_0000057)
+ObjectPropertyDomain(obo:RO_0017501 obo:UBERON_0000105)
```
Comment (cmungall) [inline src/ontology/ro-edit.owl:6639]: D/R should be on the primary direction

File: src/ontology/ro-edit.owl [hunk 4]
```diff
@@ -6612,6 +6614,31 @@ AnnotationAssertion(oboInOwl:creation_date obo:RO_0017009 "2023-03-29T17:06:06Z"
...
+AnnotationAssertion(obo:IAO_0000112 obo:RO_0017500 "Fundamentally, plant development includes a gametophyte development stage and a sporophyte development stage."@en)
+AnnotationAssertion(obo:IAO_0000115 obo:RO_0017500 "A relationship between an organism and a stage it undergoes within its full life cycle."@en)
+AnnotationAssertion(terms:contributor obo:RO_0017500 <https://orcid.org/0000-0002-1373-1705>)
+AnnotationAssertion(terms:contributor obo:RO_0017500 <https://orcid.org/0000-0002-8844-9165>)
+AnnotationAssertion(terms:date obo:RO_0017500 "2023-08-09T18:37:32Z"^^xsd:dateTime)
+AnnotationAssertion(rdfs:comment obo:RO_0017500 "This relation accommodates social, behavioural and biological (developmental) stages."@en)
+AnnotationAssertion(rdfs:label obo:RO_0017500 "has life cycle stage"@en)
+SubObjectPropertyOf(obo:RO_0017500 obo:RO_0000056)
+InverseObjectProperties(obo:RO_0017500 obo:RO_0017501)
+ObjectPropertyDomain(obo:RO_0017500 obo:OBI_0100026)
```
Comment (cmungall) [inline src/ontology/ro-edit.owl:6629]: Will be rewired to COB organism

File: src/ontology/ro-edit.owl (hunk 3 above)
Comment (ddooley) [reply src/ontology/ro-edit.owl:6639]: Done

File: src/ontology/ro-edit.owl [hunk 5]
```diff
@@ -44,6 +44,7 @@ Declaration(Class(obo:BFO_0000031))
 Declaration(Class(obo:CHEBI_50906))
 Declaration(Class(obo:COB_0001000))
 Declaration(Class(obo:OGMS_0000031))
+Declaration(Class(obo:PATO_0001470))
```
Comment (anitacaron) [inline src/ontology/ro-edit.owl:47]: Why this PATO term was added in the edit file?

**Review by anitacaron — CHANGES_REQUESTED** [review]

File: src/ontology/ro-edit.owl (hunk 5 above)
Comment (ddooley) [reply src/ontology/ro-edit.owl:47]: Same as other pull request.  This must have come in from a refresh to update from master branch.

Comment (anitacaron) [reply src/ontology/ro-edit.owl:47]: Ok, I'll remove it.

**Review by anitacaron — APPROVED** [review]
Technically, I approve. Still need ontology review.

**Comment by wdduncan** [comment]
The domain/range issue mentioned [above](https://github.com/oborel/obo-relations/pull/742#issuecomment-2018325533) still needs to be resolved.

**Comment by cmungall** [comment]
> has life cycle stage is a subproperty of participates in, and the range of participates in is occurrent. However, the range of has life cycle stage is life cycle stage and life cycle stage is not a subclass of occurrent

It should be, but this is a general issue with RO imports and will be addressed by COB. As it's not pertinent to this PR, I suggest making a new issue, since the D/R declarations are correct in this PR.

**Comment by ddooley** [comment]
It turns out "**in taxon**" allows for assignment (at instance and class level) of both organisms and processes/life stages to taxa, so we can express that "pupal stage in taxon Rhopalocera".  So there is no need for "has life cycle stage/life cycle stage of". 

Discussion on Aug 12 recommended that "life cycle stage of" be a narrow synonym of "in taxon".

A side issue is that "in taxon" doesn't have a reverse relation, so there's no relation to put opposite relation narrow synonym on.

**Pull request closed without merging**